swan-cern/files/swan_computing_config.py

```diff
@@ -267,47 +286,59 @@ async def _init_spark(self, pod_labels):
+    def _condor_enabled(self):
+        """
+        return True if the user has selected an HTCondor pool.
+        """
+        condor_pool = self.spawner.user_options[self.spawner.condor_pool]
+        return condor_pool != 'none'
 
-            # Define some 18 random NodePorts on the cluster for spark using V1Service
+    async def _open_ports(self, num_ports):
+        """
+        Create a service that opens the necessary ports for the computing integrations
+        (Spark, HTCondor)
+        """
+
+        computing_ports_service = "computing-ports" + "-" + self.spawner.user.name
+        notebook_container = self._get_pod_container('notebook')
+
+        try:
+            # Define `num_ports` random NodePorts on the cluster using V1Service
             service_template_ports = []
-            spark_ports_per_pod = 18
-            for port_id in range(1, spark_ports_per_pod + 1):
+            for port_id in range(1, num_ports + 1):
                 service_template_ports.append(
                     V1ServicePort(
-                        name="spark-port-" + str(port_id),
+                        name="comp-port-" + str(port_id),
                         port=port_id
                     )
                 )
             service_template = V1Service(
                 api_version="v1",
                 kind="Service",
                 metadata=V1ObjectMeta(
-                    name=spark_ports_service
+                    name=computing_ports_service
                 ),
                 spec=V1ServiceSpec(
-                    selector=pod_labels,  # attach this service to the pod with label {spark_pod_label}
+                    selector=self.pod.metadata.labels,  # attach this service to the user pod
                     ports=service_template_ports,
                     type="NodePort"
                 )
             )
 
-            # Create V1Service which allocates random ports for spark in k8s cluster
+            # Create V1Service which allocates random ports
             try:
                 # use existing if possible
-                await self.spawner.api.delete_namespaced_service(spark_ports_service, swan_container_namespace)
-                service = await self.spawner.api.read_namespaced_service(spark_ports_service, swan_container_namespace)
+                service = await self.spawner.api.read_namespaced_service(computing_ports_service, swan_container_namespace)
             except ApiException:
                 # not existing, create
                 try:
                     service = await self.spawner.api.create_namespaced_service(swan_container_namespace, service_template)
                 except ApiException as e:
-                    raise Exception("Could not create service that allocates random ports for Spark in k8s cluster: %s\n" % e)
+                    raise Exception("Could not create service that allocates random ports for computing (Spark, HTCondor) integrations: %s\n" % e)
```

Comment:
if we use "computing resources" we don't need to write their names which, in the future, might be outdated.

Reply:
Would also interpolate for consistency.

Reply:
Agreed, it can get outdated easily. I will keep it generic.

Reply:
As for string interpolation, you want that I change this instance or everything in the file?

Reply:
Done in all the document.